testing/scenario_app/ios/Scenarios/Scenarios/TextPlatformView.m

```diff
@@ -75,7 +75,7 @@ - (instancetype)initWithFrame:(CGRect)frame
                     arguments:(id _Nullable)args
               binaryMessenger:(NSObject<FlutterBinaryMessenger>*)messenger {
   if ([super init]) {
-    _textView = [[UITextView alloc] initWithFrame:CGRectMake(50.0, 50.0, 250.0, 100.0)];
+    _textView = [[UITextView alloc] initWithFrame:CGRectMake(0.0, 0.0, 250.0, 100.0)];
```

Comment:
I know you asked offline, but I'm pretty sure I did this as a way to arbitrarily position the text within the view. 

This seems like a valid use case to me and I'm not really sure why we need to restrict it. I think that if we pursue this,we should make sure the warning lands for at least a full stable cycle and we collect feedback on that before changing it.

Reply:
For example, in testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_platform_view_clippath_iPhone 8_13.0_simulator.png the text is no longer as visible.

Reply:
The PlatformView developer should not determine where the PlatformView is displayed, the frame of the PlatformView should be determined by the user of the PlatformView. (e.g. the widget tree)

PlatformView developer can determine where the subviews of PlatformView is displayed, that is allowed. In this case, they will create the PlatformView with 0 origin and create a subview inside the PlatformView to position it as they like.

This also creates an issue where in theory -- not sure what is a valid use case -- PlatformView developer can create a PlatformView that display way outside of the PlatformView widget thinks where it is displayed. It messed up the unobstructed platform view logic, which we used to determine the frame of overlays. 

So I do think it is valid to restrict this, at least to give a warning. Because this change is harmless, we can wait for some time to gather feedbacks and decided if we should revert this or add the hard restriction. 

> For example, in testing/scenario_app/ios/Scenarios/ScenariosUITests/golden_platform_view_clippath_iPhone 8_13.0_simulator.png the text is no longer as visible.

One thing we can do, to potentially keep the screenshots intact, is to have a plain UIView as the PlatformView and having the textView to be a child of the UIView, where the origin of textView is (50, 50)

Reply:
Makes sense.

Reply:
We may want to print this kind of explanation in the warning.

Reply:
....alternatively, should we just automatically wrap non-zero origin in a UI view with a zero origin?

Reply:
> ....alternatively, should we just automatically wrap non-zero origin in a UI view with a zero origin?

Yeah we can do that when we know there is no valid case of having a none-zero origin. However, it doesn't solve the problem when there is a subview of the PlatformView displaying outside of the frame. Maybe `clipToBounds` is a better solution.

Reply:
> We may want to print this kind of explanation in the warning.

This is explained in https://github.com/flutter/flutter/issues/109700, i think having a link and ask people to comment on the issue is probably sufficient.

Reply:
> ....alternatively, should we just automatically wrap non-zero origin in a UI view with a zero origin?

Yeah we can do this, this can change the behavior of the platform view so we will wait for feedbacks before implement it.